tools/cli/commands/test.js

```diff
@@ -0,0 +1,165 @@
+					type: 'string',
+				} )
+				.positional( 'test', {
+					alias: 't',
+					describe: 'The test to run',
+					type: 'Array',
+					choices: [ 'js', 'php', 'coverage' ],
+				} );
+		},
+		async argv => {
+			await testCli( argv );
+			if ( argv.v ) {
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Runs the test command
+ *
+ * @param {object} argv - the arguments being passed.
+ */
+async function testCli( argv ) {
+	// Handle choosing a project.
+	if ( ! argv.project || argv.project === '' ) {
+		argv = await promptForProject( argv );
+	}
+
+	argv = await validateProject( argv );
+	// Handle choosing a test type.
+	if ( ! argv.test || argv.test === '' ) {
+		argv = await promptForTest( argv );
+	}
+
+	// Get the test script for the project.
+	argv.testScript = await getTestInstructions( argv );
+
+	runTest( argv );
+}
+
+/**
+ * Validate the project we're being passed.
+ *
+ * @param {object} argv - the arguments being passed.
+ * @returns {object} argv
+ */
+async function validateProject( argv ) {
+	if ( allProjects().includes( argv.project ) ) {
+		return argv;
+	}
+	console.log( chalk.red( 'Invalid project. Reverting to interactive mode' ) );
+	argv.project = '';
+	argv = await promptForProject( argv );
+	return argv;
+}
+
+/**
+ * Gets list of tests available for chosen projects.
+ *
+ * @param {string} project - the project we want tests for..
+ * @returns {object} argv.
+ */
+async function getTests( project ) {
+	const composerJson = await readComposerJson( project );
+	if ( ! composerJson.scripts ) {
+		console.log( chalk.red( `No tests found in ${ project }'s composer.json file!` ) );
+		process.exit();
+	}
+
+	let tests = Object.keys( composerJson.scripts ).filter( test => test.startsWith( 'test-' ) );
+	tests = tests.map( test => test.substring( 5 ) );
+	if ( tests.length === 0 ) {
+		console.log( chalk.red( `No tests found in ${ project }'s composer.json file!` ) );
+		process.exit();
+	}
+	return tests;
+}
+
+/**
+ * Gets the test instructions required for the project.
+ *
+ * @param {object} argv - the arguments passed.
+ * @returns {Array} testScript - array containing test scripts.
+ */
+async function getTestInstructions( argv ) {
+	switch ( argv.test ) {
+		case 'js':
+			return 'test-js';
+		case 'php':
+			return 'test-php';
+		case 'coverage':
+			return 'test-coverage';
+		default:
+			// shouldn't ever happen, but being safe.
+			console.log(
+				chalk.red( `No ${ argv.test } script located in ${ argv.project }'s composer.json file!` )
+			);
+			process.exit();
+	}
+}
+
+/**
+ * Runs the test script.
+ *
+ * @param {object} argv - the arguments passed.
+ */
+async function runTest( argv ) {
+	console.log( chalk.green( `Running ${ argv.testScript } tests for ${ argv.project }` ) );
+	child_process.spawnSync( 'composer', [ 'run', argv.testScript ], {
+		stdio: 'inherit',
+		cwd: path.resolve( `projects/${ argv.project }` ),
+	} );
```

Comment:
You should relay the exit code from the subprocess.
```suggestion
	const res = child_process.spawnSync( 'composer', [ 'run', argv.testScript ], {
		stdio: 'inherit',
		cwd: path.resolve( `projects/${ argv.project }` ),
	} );
	process.exitCode = res.status;
```